test/ibmq/test_account_client.py

```diff
@@ -248,6 +250,42 @@ def test_job_cancel(self):
                     # just retry.
                     self.assertIn('JOB_NOT_CANCELLED', str(ex))
 
+    def test_access_token_not_in_exception_chain(self):
+        """Check that access token is replaced within chained request exceptions."""
+        backend = self.provider.backends.ibmq_qasm_simulator
+        circuit = transpile(self.qc1, backend, seed_transpiler=self.seed)
+        qobj = assemble(circuit, backend, shots=1)
+
+        api = backend._api
+        job = backend.run(qobj)
+        job_id = job.job_id()
+
+        session_ = api.client_api.session
+        access_token_ = session_.access_token
+
+        exception_message = 'The access token in this exception ' \
+                            'message should be replaced: {}'.format(access_token_)
+        self.assertIn(access_token_, exception_message)
+
+        with self.assertRaises(RequestsApiError) as exception_cm:
+            with mock.patch.object(
+                    HTTPConnectionPool,
+                    'urlopen',
+                    side_effect=MaxRetryError(
+                        HTTPConnectionPool('host'), 'url', reason=exception_message)):
+                _ = api.job_get(job_id)
+
+        def _assert_token_not_in_exc_chain(test_class, access_token, exc):
+            """Helper function to assert access token not in exception chain."""
+            if exc.__cause__:
+                _assert_token_not_in_exc_chain(test_class, access_token, exc.__cause__)
+            elif exc.__context__:
```

Comment:
Can you make your verification closer to what a user would see (e.g. stderr)? If the verification uses the same code as mainline, then it doesn't really help much. For example, if `exc` has another attribute `__foo__` that also contains the access_token, both mainline and this test would fail to catch that.

Reply:
Yeah, that would make sense. Instead of relying on the same mechanism to check whether the access token is removed from the exception chain, I decided to check the entire traceback instead, since this it is what the user would see.